pkg/pod/status.go

```diff
@@ -147,15 +152,26 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 	return *trs, merr.ErrorOrNil()
 }
 
-func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun) *multierror.Error {
+func setTaskRunStatusBasedOnStepStatus(ctx context.Context, logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun, kubeclient kubernetes.Interface) *multierror.Error {
 	trs := &tr.Status
 	var merr *multierror.Error
 
+	sidecarLogResults := []v1beta1.PipelineResourceResult{}
+	sidecarLogsResultsEnabled := config.FromContextOrDefaults(ctx).FeatureFlags.ResultExtractionMethod == config.ResultExtractionMethodSidecarLogs
+	if sidecarLogsResultsEnabled && tr.Status.TaskSpec.Results != nil {
+		resultSidecarName := fmt.Sprintf("sidecar-%s", sidecarlogsvalidation.ReservedResultsSidecarName)
```

Comment:
I find it confusing that there's a reserved sidecar name defined elsewhere, but then it's modified (prefix added) here: 

https://github.com/tektoncd/pipeline/blob/f9021b1385c0d5bfa4e61b9473597ca03d173cb6/internal/sidecarlogsvalidation/sidecarlogs.go#L20-L22

Reply:
I think when a container is created from the sidecar name, `sidecar-` prefix is added by default. So the name of the sidecar is `tekton-log-results` but the name of the container is `sidecar-tekton-log-results`.

Reply:
I could also create another reserved const `ReservedResultsSidecarContainerName`.
How about that?

Reply:
I could also name the variable better to avoid confusion.

Reply:
Yes please, that would be clearer!

Reply:
or rename the variable `resultSidecarName` to `resultSidecarContainerName`

Reply:
I created a new const `ReservedResultsSidecarContainerName` and I use that to avoid confusion. Thanks for pointing this out!